plugin/callgraph/callgraph.go

```diff
@@ -128,21 +144,34 @@ func (cg *CallGraph) addRootNode(treeNode *sitter.Node) {
 	cg.RootNode = cg.Nodes[cg.FileName]
 }
 
-func (cg *CallGraph) AddNode(identifier string, treeNode *sitter.Node) {
-	if _, exists := cg.Nodes[identifier]; !exists {
+func (cg *CallGraph) ensureNodeInAssignmentGraph(identifier string, treeNode *sitter.Node) {
+	cg.assignmentGraph.AddNode(identifier, treeNode)
+}
+
+func (cg *CallGraph) AddNode(identifier string, treeNode *sitter.Node) *CallGraphNode {
```

Comment:
Revisit if this need to be public / exported API or internal to `CallGraph`. Also look at concurrency needs.

Reply:
yep, we don't need it exposed for now